### PR TITLE
Static Terms with new GC

### PR DESCRIPTION
**Pull request by gtrepta**

### CODE REVIEW

**Comment by dwightguth** [comment]
I'd really actually like to see this branch have a merge commit rather than cherry picking the commit from master. That will keep the history much cleaner once this branch is merged back into master.

**Comment by gtrepta** [comment]
@dwightguth It's a merge commit now.

**Review by dwightguth — APPROVED** [review]
I'm approving. Please don't merge, I'm going to do that on the command line in order to preserve the history.

**Pull request merged**
